### ISSUE TITLE
mpy_ld.py: Unable to emit relocations in more complex .rodata.* sections

### ISSUE THREAD

**Issue by phlash** [micropython/micropython#8783]
While linking a large(ish) project with the latest micropython and ESP-IDF compiler, I had a failure during writing of relocations in `build_mpy` due to a section named `.rodata.str1.1` which was not matched by the `kind` checking logic, resulting in an attempt to concatenate an `int` with a `str`. Given such sections are part of the general read-only data, I applied the following fix which seems to work:
```diff
@@ -965,7 +976,7 @@ def build_mpy(env, entry_offset, fmpy, native_qstr_vals, native_qstr_objs):
     for base, addr, kind in env.mpy_relocs:
         if isinstance(kind, str) and kind.startswith(".text"):
             kind = 0
-        elif kind in (".rodata", ".data.rel.ro"):
+        elif isinstance(kind, str) and (kind.startswith(".rodata") or kind.startswith(".data.rel.ro")):
             if env.arch.separate_rodata:
                 kind = rodata_const_table_idx
             else:
```

**Comment by jonnor** [comment]
I hit the same issue when working on https://github.com/emlearn/emlearn-micropython - which was some quite simple 100 line code. I did some other hack, but will try to test this fix.

**Comment by jonnor** [comment]
This fix looks to work for me. I have created a merge request with your fix @phlash - hope you dont mind

**Comment by phlash** [comment]
You're welcome, I didn't have a fork to commit to / issue a PR from, so thanks!

**Comment by dpgeorge** [comment]
Fixed by 4837ec336a8047a1707534315f6b4b8aeac3f891
